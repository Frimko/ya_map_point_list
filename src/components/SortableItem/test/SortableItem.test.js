import React from 'react';
import { mount, shallow } from 'enzyme';
import { SortableItem } from '../index';

describe('SortableItem component', () => {
  let item = {
    id: '6565656543',
    name: 'Это метка1',
  };

  describe('snapshot', () => {
    test('snapshot', () => {
      const onDelete = jest.fn().mockName('delete');
      expect(shallow(<SortableItem item={item} deleteMark={onDelete} />)).toMatchSnapshot();
    });
  });

  describe('functional test - click to delete icon', () => {
    test('onDelete', () => {
      const onDelete = jest.fn().mockName('delete');
      const container = mount(<SortableItem item={item} deleteMark={onDelete} />);

      const icon = container.find('i');
      icon.simulate('click');
      expect(onDelete).toBeCalledWith(item.id);
    });
  });
});
