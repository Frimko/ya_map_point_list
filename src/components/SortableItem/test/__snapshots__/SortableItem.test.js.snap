// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableItem component snapshot snapshot 1`] = `
<Item
  actions={
    Array [
      <Icon
        onClick={[Function]}
        type="close"
      />,
    ]
  }
>
  <div>
    Это метка1
  </div>
</Item>
`;
